op-e2e/actions/proofs/operator_fee_test.go

```diff
@@ -0,0 +1,147 @@
+		env.Miner.ActL1IncludeTx(env.Dp.Addresses.Deployer)(t)
+		env.Miner.ActL1EndBlock(t)
+
+		// sequence L2 blocks, and submit with new batcher
+		env.Sequencer.ActL1HeadSignal(t)
+		env.Sequencer.ActBuildToL1Head(t)
+		env.Batcher.ActSubmitAll(t)
+		env.Miner.ActL1StartBlock(12)(t)
+		env.Miner.ActL1EndBlock(t)
+
+		aliceInitialBalance, err := env.Engine.EthClient().
+			BalanceAt(context.Background(), env.Alice.Address(), nil)
+		require.NoError(t, err)
+
+		env.Sequencer.ActL2StartBlock(t)
+		// Send an L2 tx
+		env.Alice.L2.ActResetTxOpts(t)
+		env.Alice.L2.ActSetTxToAddr(&env.Dp.Addresses.Bob)
+		env.Alice.L2.ActMakeTx(t)
+		env.Engine.ActL2IncludeTx(env.Alice.Address())(t)
+		env.Sequencer.ActL2EndBlock(t)
+
+		receipt := env.Alice.L2.LastTxReceipt(t)
+
+		// Check that the operator fee was applied
+		require.Equal(t, OperatorFeeScalar, uint32(*receipt.OperatorFeeScalar))
+		require.Equal(t, OperatorFeeConstant, *receipt.OperatorFeeConstant)
+
+		l1FeeVaultBalance, err := env.Engine.EthClient().BalanceAt(context.Background(), predeploys.L1FeeVaultAddr, nil)
+		require.NoError(t, err)
+
+		baseFeeVaultBalance, err := env.Engine.EthClient().
+			BalanceAt(context.Background(), predeploys.BaseFeeVaultAddr, nil)
+		require.NoError(t, err)
+
+		sequencerFeeVaultBalance, err := env.Engine.EthClient().BalanceAt(context.Background(), predeploys.SequencerFeeVaultAddr, nil)
+		require.NoError(t, err)
+
+		operatorFeeVaultBalance, err := env.Engine.EthClient().BalanceAt(context.Background(), predeploys.OperatorFeeVaultAddr, nil)
+		require.NoError(t, err)
+
+		aliceFinalBalance, err := env.Engine.EthClient().BalanceAt(context.Background(), env.Alice.Address(), nil)
+		require.NoError(t, err)
+
+		require.True(t, aliceFinalBalance.Cmp(aliceInitialBalance) < 0, "Alice's balance should decrease")
+
+		// Check that the operator fee sent to the vault is correct
+		require.Equal(t,
+			new(big.Int).Add(
+				new(big.Int).Div(
+					new(
+						big.Int,
+					).Mul(new(big.Int).SetUint64(receipt.GasUsed), new(big.Int).SetUint64(uint64(OperatorFeeScalar))),
+					new(big.Int).SetUint64(1e6),
+				),
+				new(big.Int).SetUint64(OperatorFeeConstant),
+			),
+			operatorFeeVaultBalance,
+		)
+
+		// Check that no Ether has been minted or burned
+		// All vault balances are 0 at the beginning of the test
+		finalTotalBalance := new(big.Int).Add(
+			aliceFinalBalance,
+			new(big.Int).Add(
+				new(big.Int).Add(l1FeeVaultBalance, sequencerFeeVaultBalance),
+				new(big.Int).Add(operatorFeeVaultBalance, baseFeeVaultBalance),
+			),
+		)
+
+		require.Equal(t, aliceInitialBalance, finalTotalBalance)
+
+		// Check that the difference in alice's balance is equal to the total fee
+		aliceBalanceDiff := new(big.Int).Sub(aliceInitialBalance, aliceFinalBalance)
+		require.Equal(t,
+			aliceBalanceDiff,
+			new(big.Int).Add(
+				new(big.Int).Add(l1FeeVaultBalance, operatorFeeVaultBalance),
+				new(big.Int).Add(sequencerFeeVaultBalance, baseFeeVaultBalance),
+			),
+		)
```

Comment:
Isn't this check the same as the one above? It seems superfluous to me.

Reply:
Done